### ISSUE TITLE
[$1000] Web - Assign task - Undefined appears if task title is 0

### ISSUE THREAD

**Issue by izarutskaya** [Expensify/App#25588]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. go to any chat
2. click + button and select assign task option
3. enter 0 as title and click on Next button
3. click on confirm task button
4. check the task tile on shared user

## Expected Result:
should show 0 as task title

## Actual Result:
undefined appears instead of 0

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** v1.3.55-7

**Reproducible in staging?:** Y

**Reproducible in production?:** Y

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/115492554/8f498289-b510-4ae5-a5ea-24939df795d5

https://github.com/Expensify/App/assets/115492554/aa1512ad-8233-467c-9b78-0210216893f5

**Expensify/Expensify Issue URL:**

**Issue reported by:** @gadhiyamanan

**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1691753089054979

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01e3e0cc6a5e5cea69</li>
        <li>Upwork Job ID: 1694801608943554560</li>
        <li>Last Price Increase: 2023-08-28</li>
    </ul>
</details>

**Comment by DylanDylann** [comment]
I think it is a BE bug, when the task title is 0, BE don't return childReportName field

**Comment by maddylewis** [comment]
heya @mollfpr - i _think_ this needs to be internal. but lmk if i'm mistaken and i can add the external label. thx!

**Comment by maddylewis** [comment]
adding external label

**Comment by jjcoffee** [comment]
@maddylewis This is a BE bug, see @DylanDylann's [comment](https://github.com/Expensify/App/issues/25588#issuecomment-1686440682).

**Comment by jjcoffee** [comment]
@maddylewis Actually scratch that, I can't reproduce on prod v1.3.56-24 or staging v1.3.57-5, so I think this must have been fixed!

**Comment by maddylewis** [comment]
nice! thanks for confirming :)

**Comment by gadhiyamanan** [comment]
Issue is still reproducible. if you haven't opened task thread from receiver side it appears as undefined https://github.com/Expensify/App/assets/54790231/3046a67f-255d-405d-a6d0-7455ac84cd77

cc: @maddylewis @jjcoffee

**Comment by jjcoffee** [comment]
Open to proposals!

**Comment by studentofcoding** [comment]
# Proposal

## Please re-state the problem that we are trying to solve in this issue
Web - Assign task - Undefined appears if task title is 0

## What is the root cause of that problem?
0 is used in a string context, it is considered as a falsy value. This means that if the task title is 0, it will be treated as if it's an empty string or undefined.

This happen on `const reportName = (title || report.reportName).trim();` on `editTaskAndNavigate` function 
https://github.com/Expensify/App/blob/1eafc7dff1cb609e72a97d6f1d1993a2bf22aaa6/src/libs/actions/Task.js#L404-L409

## What changes do you think we should make in order to solve the problem?
Check if `title` is undefined, thus ensuring that 0 is treated as a valid title.
```
const reportName = (_.isUndefined(title) ? report.reportName : title).trim();
```

## What alternative solutions did you explore? (Optional)

None

cc: @jjcoffee @maddylewis

**Comment by b4s36t4** [comment]
I think this is backend issue. When we create a new task backend should send `reportAction` with `childReportName` which we're using here.

https://github.com/Expensify/App/blob/5f9bb2e1d901eb6dc0110d8214e679d4d2aa6822/src/components/ReportActionItem/TaskPreview.js#L70


https://github.com/Expensify/App/assets/59088937/6183ae0a-7e80-4908-bfff-1b15c380304c

Same issue happens in multiple places like MoneyRequestDescription page etc.

@mountiny This is same as the issue you've asked about `JSON.stringify` method. 

This also can be solved if we `JSON.stringify` when sending the title to backend but that still requires to use `parse` method in multiple places.

**Comment by napster125** [comment]
I completely agree with @b4s36t4's perspective. It's evident that this issue is rooted in the backend. @studentofcoding, it's important to note that `title` is a string, not a number. Therefore, "0" should not be treated as a falsy value in JavaScript. In my opinion, the backend seems to be mishandling the parsing of "0."

Additionally, it's worth mentioning that the function `editTaskAndNavigate` doesn't appear to be related to this problem. This issue is related to the creation of a task, not its editing.

To sum it up, this seems to be a backend issue stemming from the incorrect parsing of "0" in the `title` field, and it needs to be addressed at the backend level.

**Comment by studentofcoding** [comment]
Yes, I found the same behavior as @b4s36t4, and I also slipped on mentioning `createTaskAndNavigate` instead of  `editTaskAndNavigate`.

Nevertheless, It's on the backend level and should managed first

**Comment by Pluto0104** [comment]
Hey @studentofcoding, there doesn't appear to be any issues with the frontend.

**Comment by jjcoffee** [comment]
@maddylewis Can you make this one internal? As [others](https://github.com/Expensify/App/issues/25588#issuecomment-1708771725) have pointed out this is a BE issue as Pusher event does not contain a `childReportName` in this case.

**Comment by mountiny** [comment]
@maddylewis I think you will have to add Engineering label as well to get internal assignee

**Comment by maddylewis** [comment]
thanks @mountiny for the reminder. sorry for the delay on my end!!! 

@neil-marcellini - added eng label since this is internal 👍

**Comment by b4s36t4** [comment]
@maddylewis I think this have to be internal if we don't want multiple changes in frontend. As per the above comments you should make it internal and engineering label.

**Comment by jjcoffee** [comment]
@maddylewis This one's internal - can you remove the `Help Wanted` label too?

**Comment by maddylewis** [comment]
>  you should make it internal and engineering label.

yep, this was made internal/engineering label was added last week / Monday

**Comment by neil-marcellini** [comment]
I added this to my list of weekly issues as the 3rd most important one.

**Comment by neil-marcellini** [comment]
Ok I felt bad about letting this drag on longer. After 1 hour of debugging I found the 1 line fix on the backend 😅

**Comment by maddylewis** [comment]
is this resolved? https://github.com/Expensify/Web-Expensify/pull/39198

i don't think there is any payment due for this one since it was worked on by an internal/contractor engineers, but lmk - ty!

**Comment by maddylewis** [comment]
just confirming if there's any thing left to do on this one before closing! https://github.com/Expensify/App/issues/25588#issuecomment-1808637007

**Comment by jjcoffee** [comment]
I think @gadhiyamanan may be due payment for the bug report? Otherwise nothing else!

**Comment by maddylewis** [comment]
the upwork job was no longer open so i created a new job - $250 reporting bonus sent to @gadhiyamanan. once the offer is accepted, i will close this out.

**Comment by gadhiyamanan** [comment]
@maddylewis offer accepted, thanks!
